### PR TITLE
MM-37187 Update react-intl and update localized icons to support that

### PR DESCRIPTION
**Pull request by hmhealey**
Updating react-intl forced us to fix a bunch of places where we somewhat misused `FormattedMessage`. Technically, it worked, but the typing has gotten stricter now, so that starts throwing type errors.

I originally went for using `useIntl` and `localizeMessage` in most places because I forgot I had added `LocalizedIcon` a while back. There's a few places where I left those because it was just easier to do there, but I might have forgotten to update some others.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-37187

#### Related Pull Requests
https://github.com/mattermost/mattermost-webapp/pull/8861

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by calebroseland — APPROVED** [review]

**Review by michelengelen — APPROVED** [review]
only minor things here ... nothing really blocking

File: components/file_upload_overlay.tsx [hunk 1]
```diff
@@ -29,17 +31,10 @@ const FileUploadOverlay: React.FC<Props> = (props: Props) => {
...
-                                <i
-                                    className='fa fa-upload'
-                                    title={title as string}
-                                />
-                            )}
-                        </FormattedMessage>
+                        <i
+                            className='fa fa-upload'
+                            title={formatMessage({id: 'generic_icons.upload', defaultMessage: 'Upload Icon'})}
+                        />
```
[the remark this answers is not in the document]
Comment (hmhealey) [reply components/file_upload_overlay.tsx:37]: Missed that one. Good catch!

File: components/localized_icon.tsx [hunk 2]
```diff
@@ -7,33 +7,45 @@ import {PrimitiveType, FormatXMLElementFn} from 'intl-messageformat';
...
-            defaultMessage,
-            values,
-        },
+        component = 'i',
+        ariaLabel,
+        title,
         ...otherProps
     } = props;
 
+    // Use an uppercase name since React thinks anything lowercase is an HTML tag
```
[the remark this answers is not in the document]
Comment (hmhealey) [reply components/localized_icon.tsx:26]: I know they're separate things, but I still haven't figured out where JSX is used without React 😛

**Comment by hmhealey** [comment]
@jgilliam17 I'm not sure how much there is to actually test here since the actual code changes change the tooltips and hidden aria-labels for various icons throughout the app

**Review by jgilliam17 — APPROVED** [review]
Thanks @hmhealey 
Tested, looks good to merge. 
- Verified tooltips for various icons throughout the app - no change, all display as before.

**Pull request merged**